Allow short ternary (Elvis) operator in codebase

pfefferle opened the pull request:
Removes all inline phpcs ignore comments for the Universal.Operators.DisallowShortTernary rule and updates phpcs.xml to exclude this rule, permitting use of the short ternary (Elvis) operator throughout the codebase.

## Proposed changes:
- Updated PHPCS configuration to allow short ternary operators
- Removed 11 `phpcs:ignore` comments across 8 files where the Elvis operator is used

### Other information:

- [ ] Have you written new tests for your changes, if applicable?

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to '..'
*

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger a GitHub workflow that will create and push the entry into the branch. -->

<!-- Due to org permissions, the job may fail for PRs crated from a fork under GitHub organizations. -->
<!-- In this case, you can create entry manually with `composer changelog:add` and push it into the branch. -->

<!-- If no changelog entry is required for this PR, please add the "Skip Changelog" label. -->

-   [ ] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [ ] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

<!-- Add a changelog message here -->

</details>

pfefferle committed:
Allow short ternary (Elvis) operator in codebase

Removes all inline phpcs ignore comments for the Universal.Operators.DisallowShortTernary rule and updates phpcs.xml to exclude this rule, permitting use of the short ternary (Elvis) operator throughout the codebase.

obenland commented on phpcs.xml:
```diff
@@ -31,4 +31,9 @@
 		</properties>
 	</rule>
+
+	<!-- Allow short ternary operators (Elvis operator) -->
```
```suggestion
```
This is pretty self-evident. And if we can keep Elvis out of the codebase, that's an added benefit. 🙂

pfefferle committed:
Merge branch 'trunk' into allow-short-ternary

pfefferle committed:
Remove unnecessary PHPCS ignore comments for short ternary

Deleted redundant 'phpcs:ignore Universal.Operators.DisallowShortTernary.Found' comments from multiple files after excluding the rule in phpcs.xml. This cleans up the code and maintains consistency with the project's coding standards configuration.

pfefferle commented:
@obenland btw. is the short-ternary operator allowed on WP.com?

pfefferle commented:
It is not allowed on WP.com, so we have to keep it as is.

obenland commented:
> It is not allowed on WP.com, so we have to keep it as is.

We could add an exception for the activitypub plugin folder in the global phpcs.xml on Dotcom. There's precedent.

obenland committed:
Consolidate phpcs.xml configuration

- Consolidate duplicate WordPress ruleset references into single block with exclusion
- Remove standalone PHPCompatibility rule (included in PHPCompatibilityWP)
- Reorganize config values for better readability

pfefferle committed:
Merge branch 'trunk' into allow-short-ternary
